### PR TITLE
Update note about modern browsers and refcounting for GC

### PR DESCRIPTION
**Pull request by tomayac**
### Description

WebKit still uses it: https://docs.webkit.org/Deep%20Dive/MemoryManagement.html. 

### Motivation

Correct a false statement.

### CODE REVIEW

**Comment by Josh-Cena** [comment]
Hi! This page is about JavaScript engines, so perhaps we should change "browser" to "engine". In particular, the documentation you provided is for WebCore, not JavaScriptCore:

> Most of WebCore objects are not managed by JavaScriptCore’s garbage collector. Instead, we use [reference counting](https://en.wikipedia.org/wiki/Reference_counting).

...And it's easier to implement with ref counting because it's all C++-land and you can just handle circular structures specially, as that page already talked about.

Is it confusing to a significant number of readers? Would it help if this PR just change "browser" to "engine"?

File: files/en-us/web/javascript/memory_management/index.md [hunk 1]
```diff
@@ -101,7 +101,7 @@ In this context, the notion of an "object" is extended to something broader than
 
 ### Reference-counting garbage collection
 
-> **Note:** no modern browser uses reference-counting for garbage collection anymore.
+> **Note:** most modern browsers don't use reference-counting for garbage collection anymore.
```
Comment (tomayac) [inline files/en-us/web/javascript/memory_management/index.md:104]:
```suggestion
> **Note:** most modern JavaScript engines don't use reference-counting for garbage collection anymore.
```

**Comment by tomayac** [comment]
(Sorry, was OoO for a couple of days.)

Maybe changing to JS engines and weakening to "most" can work? WDYT?

**Comment by Josh-Cena** [comment]
I think we should be assertive and say "all", because it's indeed obsolete and anyone using it would run into troubles.

**Comment by tomayac** [comment]
SGTM, then let's just swap "browser" for "JS engine".

**Review by Josh-Cena — APPROVED** [review]
LGTM. Thanks for the update—I learned about WebCore internals today 😊

**Pull request merged**